feat(Alarm) : 알람기능 구현 및 테스트

Kim-SeonBeom opened the pull request:
## 변경사항
- #61 
- 
## 체크 리스트 (Checklist)

pull request 전에 아래 체크 리스트들을 만족하는 지 확인한 후 체크('x') 표시를 해주시기 바랍니다.

- [x]  master 브랜치가 아니라 **develop** 브랜치에 Merge하도록 pull request를 작성 중이신가요?
- [x]  모든 **테스트**가 성공했나요?

Commit:
feat(Alarm) : 알람생성, 내 알람 조회 기능 구현 및 테스트코드 작성
 1. alarmEntity 및 ErrorCode 생성
 2. 알람 생성 기능 구현(비지니스 로직으로만 존재) 및 테스트코드 작성
 3. 내 알람 조회 기능 구현 및 테스트 코드 작성

Commit:
feat(Alarm) : 알람 읽음 설정 구현 및 테스트코드 작성
 1. alarmEntity에 setAsRead 메서드 추가
 2. 알람 읽음 설정 기능 구현 및 테스트코드 작성

Commit:
feat(Alarm) : URL 수정,알람 삭제 기능 및 테스트코드 작성
 1. URL수정
 2. 알람 삭제 기능 및 테스트코드 작성

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/alarm/model/Alarm.java:
```diff
@@ -0,0 +1,26 @@
+    Long id;
+    String contents;
+    boolean isRead;
+    Long userId;
```
연관 관계로 시도해보셔도 좋을거 같아요

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/alarm/model/dto/AlarmResponseDto.java:
```diff
@@ -0,0 +1,25 @@
+@NoArgsConstructor
+@Builder
+public class AlarmResponseDto {
+    Long id;
```
private 추가 부탁드려요

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/alarm/service/AlarmService.java:
```diff
@@ -0,0 +1,51 @@
+            throw new AlarmException(AlarmErrorCode.ALARM_IS_EMPTY);
+        }
+        List<AlarmResponseDto> responseAlarmDtoList = new ArrayList<>();
+        for (Alarm alarm : alarmsList) {
```
Stream으로 만들어도 깔끔하겠네요

Commit:
fix(Alarm) : 캐시 적용, 수정사항에 따른 테스트코드 변경
1. cacheManager 설정 추가
2. addAlarm 메서드 void 타입으로 변경
3. getMyAlarms 메서드 캐시사용
4. userId를 한번더 검증하던 로직 제거
5. 3번사항에 따른 테스트코드 변경
6. ResponseEntity 와일드카드 변경

Kim-SeonBeom committed:
Merge branch 'dev' into #61-Alarm
